src/components/header/components/basketPopover/basketPopover.tsx

```diff
@@ -16,16 +16,28 @@ type Item = {
 
 interface BasketPopoverProps {
   defaultOpen: boolean;
-  basketItems: Item[];
+  handleDeleteItem?: () => void;
+  basketQuantity?: number;
+  basketItems?: Item[];
 }
 
 export function BasketPopover({
   defaultOpen,
   basketItems,
+  handleDeleteItem,
+  basketQuantity,
 }: BasketPopoverProps) {
   const [open, setOpen] = useState(defaultOpen);
 
-  const ref = useClickAway<HTMLDivElement>(() => setOpen(false));
+  const ref = useClickAway<HTMLDivElement>((e) => {
+    setOpen(false);
+  });
+
+  const handleOnBasketClick = () => {
+    if (open === false) {
+      setOpen(true);
```

Comment:
`setOpen((oldState) => {
  if (!oldState) {
    return true
  }
  return oldState
})`

Reply:
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state